src/components/producerItems/components/ProductEditModal.js

```diff
@@ -93,41 +99,96 @@ class ProductEditForm extends React.Component {
         const { form, productToEdit, updateProduct } = this.props;
         const { resetFields, validateFields } = form;
 
-        validateFields((error, fieldValues) => {
-            if (!error) {
-                updateProduct({ ...productToEdit, ...fieldValues });
-                return resetFields();
-            }
-
-            return error;
-        });
-    }
-
-    static getDerivedStateFromProps(props) {
-        const { productToEdit, form: { getFieldDecorator } } = props;
-        const formItems = generateProductEditForm(getFieldDecorator, productToEdit);
-
-        return { formItems };
-    }
-
-    render() {
-        const { closeModal, isNewProduct, isOpen, isProductUpdating } = this.props;
-        const { formItems } = this.state;
-        const title = isNewProduct ? CREATE_TITLE : EDIT_TITLE;
-
-        return (
-            <Modal
-                confirmLoading={isProductUpdating}
-                visible={isOpen}
-                title={title}
-                okText={SAVE_PRODUCT_DETAILS}
-                onCancel={closeModal}
-                onOk={this.updateProductInfo}
-            >
-                <Form>{formItems}</Form>
-            </Modal>
-        );
+        if(this.state.fileDetails.length>=1){
+            const { image_url, image_id } = this.state.fileDetails[0].response.data;
+
+            validateFields((error, fieldValues) => {
+                if (!error) {
+                    updateProduct({ ...productToEdit, ...fieldValues, image_id, image_url });
+                    return resetFields();
+                }
+                return error;
+            });
+        } else {
+            validateFields((error, fieldValues) => {
+                if (!error) {
+                    updateProduct({ ...productToEdit, ...fieldValues });
+                    return resetFields();
+                }
+                return error;
+            });
+        }
     }
+    
+      handleCancel = () => this.setState({ previewVisible: false })
+    
+      handlePreview = () => this.setState({ previewVisible: true });
```

Comment:
consolidate `handleCancel` and `handlePreview` into one function given that all there are toggling `previewVisible` between `true` and `false`

Reply:
couldn't figure out how best to consolidate these functions given that the component i used accepts different API for each func

Reply:
pass in the value
```
setPreview = value => this.setstate({ previewVisible: value });
```